### PR TITLE
Leonardo saraceli #129 create comments endpoint

### PR DESCRIPTION
**Pull request by LeonardoSaraceli**

### CODE REVIEW

**Review by MyrtheDullaart — DISMISSED** [review]

**Review by FBagdeli — DISMISSED** [review]

File: prisma/schema.prisma [hunk 1]
```diff
@@ -88,3 +90,14 @@ model Note {
   createdAt DateTime  @default(now())
   updatedAt DateTime? @updatedAt
 }
+
+model Comment {
+  id        Int       @id @default(autoincrement())
+  content   String    @db.VarChar(255)
```
Comment (vherus) [inline prisma/schema.prisma]: What was the reasoning behind setting a 255 character limit? Is this listed in the requirements?

File: src/controllers/comment.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  const { id } = req.user
+  const postId = Number(req.params.id)
+
+  const post = await postFound(postId)
+
+  if (!post) {
+    return sendDataResponse(res, 400, { error: ERR.POST_NOT_FOUND })
+  }
+
+  const comment = await createCommentDb(content, id, postId)
```
Comment (vherus) [inline src/controllers/comment.js:28]: If the requirements ask for a 255 character limit, you need to validate the data before trying to save it to the DB

File: src/controllers/comment.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    return sendDataResponse(res, 400, { error: ERR.MISSING_CONTENT })
+  }
+
+  const { id } = req.user
+  const postId = Number(req.params.id)
+
+  const post = await postFound(postId)
+
+  if (!post) {
+    return sendDataResponse(res, 400, { error: ERR.POST_NOT_FOUND })
```
Comment (vherus) [inline src/controllers/comment.js]: A not found should use the `404` status code

File: src/domain/comment.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import dbClient from '../utils/dbClient.js'
+
+const getAllCommentsDb = async () => {
+  return await dbClient.comment.findMany()
+}
+
+const postFound = async (postId) => {
```
Comment (vherus) [inline src/domain/comment.js]: This suggests it's a variable that contains a boolean value. `findPostById` would be a better name. Additionally, this function shouldn't be in the `comment` domain, but in the `post` domain

File: src/controllers/comment.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  return sendDataResponse(res, 201, { comment })
+}
+
+const getAllPostComments = async (req, res) => {
+  const postId = Number(req.params.id)
+
+  const post = await postFound(postId)
+
+  if (!post) {
+    return sendDataResponse(res, 400, { error: ERR.POST_NOT_FOUND })
```
Comment (vherus) [inline src/controllers/comment.js]: 404

**Review by vherus — CHANGES_REQUESTED** [review]

**Review by vherus — APPROVED** [review]
Great stuff :)

**Review by FBagdeli — APPROVED** [review]

**Pull request merged**